instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -299,6 +301,77 @@ def _inject_propagation_headers(headers, args, kwargs):
         kwargs["headers"] = _headers.raw
 
 
+def _extract_response(
+    response: typing.Union[
+        httpx.Response, typing.Tuple[int, Headers, httpx.SyncByteStream, dict]
+    ]
+) -> typing.Tuple[int, Headers, httpx.SyncByteStream, dict, str]:
+    if isinstance(response, httpx.Response):
+        status_code = response.status_code
+        headers = response.headers
+        stream = response.stream
+        extensions = response.extensions
+        http_version = response.http_version
+    else:
+        status_code, headers, stream, extensions = response
+        http_version = extensions.get("http_version", b"HTTP/1.1").decode(
+            "ascii", errors="ignore"
+        )
+
+    return (status_code, headers, stream, extensions, http_version)
+
+
+def _apply_request_client_attributes_to_span(
+    span_attributes, url, method_original, span_name, semconv
+):
+    # http semconv transition: http.method -> http.request.method
+    _set_http_method(span_attributes, method_original, span_name, semconv)
+    # http semconv transition: http.url -> url.full
+    _set_http_url(span_attributes, str(url), semconv)
+    try:
+        if _report_new(semconv):
+            if url.host:
+                # http semconv transition: http.host -> server.address
+                _set_http_host(span_attributes, url.host, semconv)
+                # http semconv transition: net.sock.peer.addr -> network.peer.address
+                span_attributes[NETWORK_PEER_ADDRESS] = url.host
+            if url.port:
+                # http semconv transition: net.sock.peer.port -> network.peer.port
+                _set_http_peer_port_client(span_attributes, url.port, semconv)
+                span_attributes[NETWORK_PEER_PORT] = url.port
+    except ValueError:
+        _logger.warning("Failed to parse attributes from URL: %s", url)
+
+
+def _apply_response_client_attributes_to_span(
+    span, status_code, http_version, semconv
+):
+    span_attributes = {}
+    # http semconv transition: http.status_code -> http.response.status_code
+    _set_http_status_code(
+        span_attributes,
+        status_code,
+        semconv,
+    )
+    http_status_code = http_status_to_status_code(status_code)
```

Comment:
Would you be able to use [set_status](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/opentelemetry-instrumentation/src/opentelemetry/instrumentation/_semconv.py#L344) for this part?

Reply:
Yes. I think it's possible, I remember I did it this way because I saw you were working on some changes of this function in asgi transition and I was unsure about the parameters order

Reply:
# TODO: use _set_status when it's stable for http clients